programs/lz4io.c

```diff
@@ -395,8 +404,50 @@ typedef struct {
     void*  dstBuffer;
     size_t dstBufferSize;
     LZ4F_compressionContext_t ctx;
+    LZ4F_CDict* cdict;
 } cRess_t;
 
+static void LZ4IO_loadDict(char **dictionaryBuffer, size_t *dictionarySize) {
```

Comment:
Some coding style guidelines : 
1) prefer `void*` as buffer type. Keep `char*` to indicate it's an ascii/text content (typically for `printf()`).
2) I believe that this function objective is to deliver a buffer filled with dictionary content.
In which case, deliver it by value.
Prototype becomes : 
`void* LZ4IO_createDictBuffer(size_t* dictionarySize);`
`create` is an important keyword, that states that data a memory area has been allocated. It must be paired with a `free` later in the same context.
3) This function can fail, for whatever reason. Document it.
For functions which deliver allocated buffers, it's typical to consider result `NULL` as an error.